src/javascripts/dropdown-switch.js

```diff
@@ -0,0 +1,109 @@
+      dc.style.display = 'block';
+    };
+
+    // Hide visible content blocks function
+    var contentHide = function (dc) {
+      dc.style.display = 'none';
+    };
+
+    // Set default content visibility state
+    dropdownContentList.forEach(function (elements) {
+
+      // Grab visibility state (show/hide) of each content block
+      var dropdownContentInitalState = elements.getAttribute('data-dropdown-content');
+
+      if (dropdownContentInitalState === 'show') {
+        contentShow(elements);
+      } else if (dropdownContentInitalState === 'hide') {
+        contentHide(elements);
+      }
+    });
+
+    // Loop through content switches
+    dropdownSwitcherList.forEach(function (elements) {
+
+      // Grab the switch/content holder using closestElement function
+      var dropdownHolder = closestParentElement(elements, function (el) {
+        return el.querySelector('[data-dropdown-content]');
+      });
+
+      // Grab the content block associated with holder and switch
+      var dropdownContent = dropdownHolder.querySelector('[data-dropdown-content]');
+
+      // Store reference to content data attribute
+      var dropdownContentState = dropdownContent.getAttribute('data-dropdown-content');
+
+      // Handle checkbox switches
+      if (elements.type === 'checkbox') {
+        elements.addEventListener('click', function () {
+          if (this.checked) {
+            if (dropdownContentState === 'show') {
+              contentHide(dropdownContent);
+            } else if (dropdownContentState === 'hide') {
+              contentShow(dropdownContent);
+            }
+          } else {
+            if (dropdownContentState === 'show') {
+              contentShow(dropdownContent);
+            } else if (dropdownContentState === 'hide') {
+              contentHide(dropdownContent);
+            }
+          }
+        });
+      }
+
+      // Handle radio group switches
+      if (elements.type === 'radio') {
+
+        // Grab radio group name
+        var inputSwitchName = elements.name;
+
+        // Grab radio group elements with that name
+        var inputSwitchNamesArray = document.getElementsByName(inputSwitchName);
+
+        // function to set submit button type from radio selection
+        var setRadios = function () {
+          if (this.getAttribute('data-dropdown') === 'switch' && this.checked) {
+            if (dropdownContentState === 'show') {
+              contentHide(dropdownContent);
+            } else if (dropdownContentState === 'hide') {
+              contentShow(dropdownContent);
+            }
+          } else {
+            if (dropdownContentState === 'show') {
+              contentShow(dropdownContent);
+            } else if (dropdownContentState === 'hide') {
+              contentHide(dropdownContent);
+            }
+          }
+        };
+
+        // Loop radio group and grab clicked radio button
+        for (var x = 0; x < inputSwitchNamesArray.length; x++) {
```

Comment:
you use a forEach above and a for loop here - consistency?

Reply:
I know I originally did this differently for a reason but there is also no problem using foreach - amended and resubmitted.